### ISSUE TITLE
Catching up with the updates

### ISSUE THREAD

**Issue by FurkanGozukara** [bmaltais/kohya_ss#3027]
Hello @bmaltais . I hope you are ok. Community is demanding to catching up the updates if possible. Thank you so much for your efforts

Here few list

https://github.com/kohya-ss/sd-scripts/commit/8e378cf03df645cef897a342559dc5fa7f66a35d
https://github.com/kohya-ss/sd-scripts/commit/bdf9a8cc29c47586467914f956c5e130d81e7723
https://github.com/kohya-ss/sd-scripts/commit/420a180d938c7b5a6e3006b1719dbfeaae72a2cc
https://github.com/kohya-ss/sd-scripts/commit/2cb7a6db02ae001355f4830581b9fc2ffffe01c6

These are some of them.

But mostly add the option in gui for huber_scale 
Sd3.5 training options for clip l and t5 as well
new schedulers that were added, prodigy plus/ schedule free ones
weighting_scheme options for sd3.5 and flux
block swapping option for sd3.5 as well

**Comment by bmaltais** [comment]
Yeah, this is something I have put off for a while. Adding SD3 LoRA support will be a pain.

**Comment by Sylsatra** [comment]
Yes please Progidy with p slice!

**Comment by FurkanGozukara** [comment]
> Yeah, this is something I have put off for a while. Adding SD3 LoRA support will be a pain.

awesome thank you so much

**Comment by XCogni** [comment]
@bmaltais Huber scale can be used with higher value than 1.0 you can change the max value to 10

**Comment by bmaltais** [comment]
OK, I have implemented a MVP for SD3 LoRA training in the GUI. Gove it a tray and let me know how it goes.

I don't use SD3 much so I won't be able to properly test the implementation. Please report issues if you find them.

**Comment by XCogni** [comment]
![Screenshot 2024-12-31 073114](https://github.com/user-attachments/assets/702302f5-bbaf-4da2-8455-af9146042bb4)
2024-12-31 07:30:15 INFO     clip_l is not included in the checkpoint and clip_l_path is not        sd3_utils.py:118
                             provided
                             
                             getting this error even though i put the path for the clip l model

**Comment by bmaltais** [comment]
Does it appear in the toml file under clip_l?

**Comment by XCogni** [comment]
Nope only the t5 and clip g ones show up, clip l does not

**Comment by XCogni** [comment]
also got a dtype error when trying to train under fp8

**Comment by bmaltais** [comment]
Let me check why clip_l does not show up... weird... I have been able to reproduce the issue on my system... so I can properly track the issue

**Comment by XCogni** [comment]
try to get training to start i think

**Comment by bmaltais** [comment]
OK, I got the training to start. There was a few bugs. Should somehow work now

**Comment by bmaltais** [comment]
Now I discovered a bug with the config save and load... more fixing needed ;-(

**Comment by XCogni** [comment]
Also you need to add weighting_scheme uniform as well. Sd3.5 will train better with it

**Comment by bmaltais** [comment]
OK... added. My issue with config save and load is fixed... Was just a browser GUI refresh issue.

**Comment by XCogni** [comment]
Training started! thank you for the updates, time to test out the results

**Comment by XCogni** [comment]
damn sd3.5 is just horrible

**Comment by 5KilosOfCheese** [comment]
> damn sd3.5 is just horrible

I actually like many aspects of it. However... Being used to Flux and how it works, it's hard to get used to it.  There were certain things I was able to do better with SD3.5 than Flux, and vice versa. But what I liked most about SD3.5 is that it's runs bit faster - especially when changing prompts. At least on Comfy and my computer. However... I still prefer flux.  But this is also because I mainly fuck around with  my own fine tunes and like to see what I can fine tune.

**Comment by XCogni** [comment]
@5KilosOfCheese yeah for styles it seems decent to train on, but training for realism, the anatomy and distortions are pretty bad
